src/main/java/de/rondiplomatico/nds/polygon.java

```diff
@@ -0,0 +1,264 @@
+package de.rondiplomatico.nds;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.IntStream;
+
+public class polygon {
```

Comment:
CamelCase! -> Polygon.
Also, if this is a polygon composed of NDSCoordinates, it should be named "NDSPolygon"
also, it has a main method. add a separate Class "NDSPolyFillDemo" with it, and put the polygon logic into the NDSPolygon class

Reply:
I've moved the demo away from the class - still not sure if that's organized well. Might have to revisit the files later.